api/src/main/java/org/apache/iceberg/io/ParallelIterable.java

```diff
@@ -72,8 +73,7 @@ private ParallelIterator(Iterable<? extends Iterable<T>> iterables,
             }
           }).iterator();
       this.workerPool = workerPool;
-      // submit 2 tasks per worker at a time
-      this.taskFutures = new Future[2 * ThreadPools.WORKER_THREAD_POOL_SIZE];
+      this.taskFutures = new Future[parallelism];
```

Comment:
Is it possible to base this on the size of the `ExecutorService`, rather than pass it in?

Reply:
I wanna to do this at first, but `ExecutorService` does not seem to provide the API to get the pool size. I think the reason is only part of `ExecutorService` will concern the pool size, and such as pool size of `CachedThreadPool` will be `Integer.MAX_VALUE`.

Reply:
What about `ThreadPoolExecutor#getPoolSize()`?

Reply:
This is okay, then. But it still needs to double the number of threads in the pool. Can you add back the multipler?